### ISSUE TITLE
Clip_Count=30 in Code and reported in the Paper  but unclear if for raw counts or normalized count

### ISSUE THREAD

**Issue by maxiallard** [czi-ai/transcriptformer#20]
Hi there,
First of all, congrats on the great paper and model! Big Fan :)

Second, I had an implementation question regarding the clip_count value used in the config here and for the training. In the paper, it is reported that raw values are clipped at 30 (which seems quite low for raw counts). Upon further checking the dataloader.py file I see that you are using the adata.X and not adata.raw.X frame which can be normalized counts if they are present whereas anndata.raw.X is always the unnormalised dataset. Would it be possible to clarify whether the model has been trained on normalised X data (which would make sense with a clipping value of 30) or if indeed all the raw counts are taken (and I missed this part somewhere) and clipped to 30 :) 

![Image](https://github.com/user-attachments/assets/cf1bc7aa-8316-4134-a1c9-5dc17da07d23)

Thanks a lot in advance and congrats again.
Best,
Maxime

**Comment by maxiallard** [comment]
Thanks for the quick comment James! Out of curiosity, how was 30 chosen? I am asking since most raw counts in datasets will go into the 1000's for many genese and that variability would be missed by that cap?

**Comment by jdpearce4** [comment]
We found that for our training data 30 covered most of the head/torso of the count distribution. But this value can vary dramatically for the assay type, Smart-Seq for example tends to be very high. If you're afraid of clipping your data too much you can use the normalize_to_scale option (model.data_config.normalize_to_scale if you're using the inference script) to scale the total counts down (it will scale the total counts to the value provided) in a uniform way. Otherwise you can try increasing the clip_count, which hasn't been tested (after training) but I suspect the model will still give sensible results, but note the cell embeddings may become dominated by just a few highly expressed genes.

**Comment by maxiallard** [comment]
Thanks for the explanation! Indeed the normalization and scaling was my next idea 😄! I will close the issue but thanks a lot for taking the time!

**Comment by maxiallard** [comment]
Hi James,
Sorry to reopen the issue but when downloading the provided anndata ([https://github.com/czi-ai/transcriptformer/blob/0862cf9ed633b1c929b1fbbc432132eaf8448048/src/transcriptformer/datasets.py#L103](https://github.com/czi-ai/transcriptformer/blob/0862cf9ed633b1c929b1fbbc432132eaf8448048/src/transcriptformer/datasets.py#L103)) I checked the counts of the file and it looks like anndata.raw.X and anndata.X are not the same:

```
print(adata.raw.X[0])
  (0, 6)	1212.0
  (0, 7)	228.0
  (0, 22)	273.0
  (0, 41)	1053.0
  (0, 53)	1.0
  (0, 55)	374.0
  (0, 60)	668.0
  (0, 91)	1.0
  (0, 100)	283.0
  (0, 111)	2.0
  (0, 116)	1.0
  (0, 125)	1594.0
...
```

And printing out the anndata.X :
```
print(adata.X[0])
  (0, 6)	3.3440208435058594
  (0, 7)	1.8151206970214844
  (0, 22)	1.9680418968200684
  (0, 41)	3.208707332611084
  (0, 53)	0.022301295772194862
  (0, 55)	2.24436092376709
  (0, 60)	2.7766194343566895
  (0, 91)	0.022301295772194862
  (0, 100)	1.9990673065185547
  (0, 111)	0.044116076081991196
  (0, 116)	0.022301295772194862
  (0, 125)	3.6095011234283447
```

The latter is used by your model I guess which seems normalised and hence the clipping seems not restrictive at all whereas it is for the first form. That's where my confusion come from I guess .. 

Thanks again for the help :)

**Comment by jdpearce4** [comment]
Yes, that's right. The thinking is that we don't want the cell embeddings to be dominated by a few very highly expressed genes. However, I'll admit this isn't a hyperparameter we spent much time tuning and is something we may revisit in future iterations. 

The perturbation setting is an interesting one, if you only have 2% of counts above the clip value I'd consider just trying to increase the clip value above the maximum count. If you have a much larger fraction about the clip value you'll probably want to scale them as well.

**Comment by maxiallard** [comment]
Thanks for clarifying!
Closing this again :)
